librz/core/cmd_type.c

```diff
@@ -729,6 +725,76 @@ static void set_offset_hint(RzCore *core, RzAnalysisOp *op, const char *type, ut
 	}
 }
 
+RZ_API void rz_core_list_loaded_typedefs(RzCore *core, const char *input, Sdb *TDB, RzOutputMode mode) {
+	PJ *pj = NULL;
+	if (mode == RZ_OUTPUT_MODE_JSON) {
+		pj = pj_new();
+		pj_o(pj);
+	}
+	char *name = NULL;
+	SdbKv *kv;
+	SdbListIter *iter;
+	SdbList *l = sdb_foreach_list(TDB, true);
+	ls_foreach (l, iter, kv) {
+		if (!strcmp(sdbkv_value(kv), "typedef")) {
+			if (!name || strcmp(sdbkv_value(kv), name)) {
+				free(name);
+				name = strdup(sdbkv_key(kv));
+				if (!input[1]) {
```

Comment:
this check was just used to see which output mode had to be used. You can just check `mode` instead. Also, by removing this, you can also remove the `input` argument. By doing so, you make the whole function cleaner and easier to use and understand.